### PR TITLE
lending.py: Add logging to exceptions

### PR DESCRIPTION
**Pull request by cclauss**
Fix.  `openlibrary/core/lending.py` contains many bare exceptions which make it difficult to spot the problems that are causing [`pytest openlibrary/plugins/openlibrary/tests/test_home.py`](https://travis-ci.org/github/internetarchive/openlibrary/jobs/674630273#L2046) to fail on Python 3.  This PR logs the exceptions in lending.py using
https://docs.python.org/3/library/logging.html#logging.Logger.exception

### Testing

Yes.

### CODE REVIEW

File: openlibrary/plugins/openlibrary/home.py [hunk 1]
```diff
@@ -238,6 +238,7 @@ def get_authors(doc):
     work = book.works and book.works[0]
     d.authors = get_authors(work if work else book)
     cover = work.get_cover() if work and work.get_cover() else book.get_cover()
+    logger.error("format_book_data(%s) work=%s, d.authors=%s" % (book, work, d.authors))
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/home.py:241]: Why are we logging an error here?

File: openlibrary/mocks/mock_infobase.py [hunk 2]
```diff
@@ -106,6 +106,11 @@ def quicksave(self, key, type="/type/object", **kw):
         }
         query.update(kw)
         self.save(query)
+
+        import logging
+        logger = logging.getLogger(__name__)
+        logger.error("quicksave(%s) query=%s -> %s" % (key, query, self.get(key)))
```
Comment (cdrini) [inline openlibrary/mocks/mock_infobase.py:112]: Why logging an error (or anything) here?

Comment (cclauss) [reply openlibrary/mocks/mock_infobase.py]: I was looking to see why pytests were failing.  Reverted.

File: openlibrary/plugins/openlibrary/home.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/plugins/openlibrary/home.py]: I was looking to see why pytests were failing.  Reverted.

**Comment by cdrini** [comment]
Could you resolve the conflicts on this branch? I tried to but I can't quite figure out what it's supposed to be; it looks like there were some big changes in the test-py3 file.

**Comment by cdrini** [comment]
Already seeing some errors logged!

```
Traceback (most recent call last):
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/core/lending.py", line 245, in get_availability
    content = urllib.request.urlopen(url=url, timeout=config_http_request_timeout).read()
  File "/usr/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 404: Not Found
2020-04-17 14:32:29 [28818] [openlibrary.core.lending] [ERROR] get_available([])
Traceback (most recent call last):
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/core/lending.py", line 216, in get_available
    request = urllib.request.Request(url=url)
  File "/usr/lib/python2.7/urllib2.py", line 202, in __init__
    self.__original = unwrap(url)
  File "/usr/lib/python2.7/urllib.py", line 1060, in unwrap
    url = url.strip()
AttributeError: 'list' object has no attribute 'strip'
2020-04-17 14:32:29 [28818] [openlibrary.core.lending] [ERROR] get_available([])
Traceback (most recent call last):
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/core/lending.py", line 216, in get_available
    request = urllib.request.Request(url=url)
  File "/usr/lib/python2.7/urllib2.py", line 202, in __init__
    self.__original = unwrap(url)
  File "/usr/lib/python2.7/urllib.py", line 1060, in unwrap
    url = url.strip()
```

**Comment by cdrini** [comment]
Note to self (and @mekarpeles ); let's keep on eye on this during deploy to make sure it doesn't explode the size of our log files.

**Comment by cclauss** [comment]
> 2020-04-17 14:32:29 [28818] [openlibrary.core.lending] [ERROR] get_available([])

`url` is an empty list!  Wrong datatype and no content.  A posterchild for bare exceptions.

**Comment by cclauss** [comment]
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/lending.py#L165-L166 is the problem because `compose_ia_url()` is supposed to return a url (`str`) but here is returning an empty list.   Need those Python 3 type hints ;-)

**Comment by cclauss** [comment]
Were there many of these in the logs?  How do we hunt down why this is happening?

**Pull request merged**